tests/test_dataset.py

```diff
@@ -62,14 +65,32 @@ def test_entry_updates_ilthermo_entry_data_columns_with_header(
     assert entry.data.columns == ["mock_header"]
 
 
+@pytest.mark.xfail
+def test_entry_caches_ilthermo_entry(mocker: MockerFixture, tmp_path: Path) -> None:
+    # This test is supposed to check weather anything is cached while Entry is
+    # constucted. Fails due to joblib not accepting mocker.Mock as function
+    # Mock.
+    mock_get_entry = mocker.Mock(return_value="mock_entry")
```

Comment:
`MagicMock` makes the object iterable what is required by `joblib`.

```suggestion
    mock_get_entry = mocker.MagicMock(return_value="mock_entry")
```

Surprisingly, this make test failing at retrieving `Entry`, so it looks like mocking isn't applied at all.